cmd/simulator/txs/agent.go

```diff
@@ -0,0 +1,111 @@
+// (c) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package txs
+
+import (
+	"context"
+	"fmt"
+)
+
+// TxSequence provides an interface to return a channel of transactions.
+// The sequence is responsible for closing the channel when there are no further
+// transactions.
+type TxSequence[T any] interface {
+	Chan() <-chan T
+}
+
+// Worker defines the interface for issuance and confirmation of transactions.
+// The caller is responsible for calling Close to cleanup resources used by the
+// worker at the end of the simulation.
+type Worker[T any] interface {
+	IssueTx(ctx context.Context, tx T) error
+	ConfirmTx(ctx context.Context, tx T) error
+	Close(ctx context.Context) error
+}
+
+// Execute the work of the given agent.
+type Agent[T any] interface {
+	Execute(ctx context.Context) error
+}
+
+// issueAllAgent provides a tx issuing agent that will issue all transactions immediately and then confirm them afterwards
+type issueAllAgent[T any] struct {
+	sequence TxSequence[T]
+	worker   Worker[T]
+}
+
+func NewIssueAllAgent[T any](sequence TxSequence[T], worker Worker[T]) Agent[T] {
+	return &issueAllAgent[T]{
+		sequence: sequence,
+		worker:   worker,
+	}
+}
+
+func (a issueAllAgent[T]) Execute(ctx context.Context) error {
+	txChan := a.sequence.Chan()
+
+	txs := make([]T, 0, 100)
+	for tx := range txChan {
```

Comment:
seems like there are two functions that do the same thing. One has a fixed N at 100 and one has a nonfixed N. We should keep the nonfixed N function and have the user decide N in config cli flags

Reply:
done